lib/src/serializable.dart

```diff
@@ -0,0 +1,237 @@
+
+// Any classes / methods that are going to be reflected must annotate with this
+class Reflectable {
+  const Reflectable();
+}
+
+abstract class Bridge<T> {
+  StreamController<Map> _eventController;
+
+  Bridge() {
+    _eventController = new StreamController<Map>.broadcast();
+  }
+
+  void broadcast(Map dataToSend);
+  void handleEvent(T apiCall);
+
+  Stream<Map> get eventReceived => _eventController.stream;
+}
+
+// This class is used to communicate across a WKWebView in an iOS app
+class WKWebViewBridge extends Bridge<CustomEvent> {
+  WKWebViewBridge() {
+    window.on['bridge'].listen(handleEvent);
+  }
+
+  @override
+  void broadcast(Map dataToSend) {
+    wkBridge?.callMethod('postMessage', [new JsObject.jsify(dataToSend)]);
+  }
+
+  @override
+  void handleEvent(CustomEvent apiCall) {
+    _eventController.add(apiCall.detail);
+  }
+
+  JsObject get wkBridge {
+    if (context['webkit'] != null) {
+      return context['webkit']['messageHandlers']['bridge'];
+    } else {
+      return null;
+    }
+  }
+}
+
+class SerializableEvent<T> extends Event<T> {
+  SerializableEvent(this._eventKey, DispatchKey dispatchKey)
+      : super(dispatchKey);
+
+  String _eventKey;
+  String get eventKey => _eventKey;
+}
+
+abstract class SerializableEvents {
+  List<SerializableEvent> get allEvents => [];
+}
+
+abstract class SerializableModule extends Module {
+  SerializableModule() {
+    _registerWithSerializableBus();
+  }
+
+  void _registerWithSerializableBus() {
+    SerializableBus.sharedBus.registerModule(this);
+  }
+
+  @override
+  SerializableEvents get events => null;
+
+  String get serializableKey => null;
+}
+
+class SerializableBus {
+  SerializableBus() {
+    _registeredModules = new Map<String, SerializableModule>();
+    _logger = new Logger('Serializable Bus');
+  }
+
+  static final SerializableBus _singleton = new SerializableBus();
+  static SerializableBus get sharedBus => _singleton;
+
+  Bridge _bridge;
+  Logger _logger;
+  Map<String, SerializableModule> _registeredModules;
+
+  void reset() {
+    _registeredModules = new Map<String, SerializableModule>();
+    _bridge = null;
+  }
+
+  void registerModule(SerializableModule module) {
+    if (module.serializableKey != null) {
+      _registerForLifecylceEvents(module);
+      _registeredModules[module.serializableKey] = module;
+    } else {
+      _logger.warning('Unable to serialize module without serializableKey');
+    }
+  }
+
+  void deregisterModule(SerializableModule module) {
+    _registeredModules.remove(module.serializableKey);
+  }
+
+  void _registerForAllEvents(SerializableModule module) {
+    if (module.events != null) {
+      for (var event in module.events.allEvents) {
+        if (event is SerializableEvent) {
+          event
+              .listen((payload) => _sendEvent(module, event.eventKey, payload));
+        }
+      }
+    } else {
+      _logger
+          .warning('Events not defined for ${module.serializableKey} module');
```

Comment:
I think `info` would be more appropriate here - modules aren't required to define any events.